Materials.xml converter

AlexanderVeselov opened the pull request.

AlexanderVeselov committed:
Fix embed kernels in uberv2

AlexanderVeselov committed:
Format script

AlexanderVeselov committed:
Fix linux build

AlexanderVeselov committed:
Move embed kernels header to build folder, add dependency on kernel files

AlexanderVeselov committed:
Small fixes

AlexanderVeselov committed:
Add path to header names, add recursive file search in stringify script

AlexanderVeselov committed:
Fix directory

AlexanderVeselov committed:
Change working directory

DenisBirukov committed:
More GLTF related fixes

-Add suffix "64" to RprSupport/RadeonProRender libs
-Add more functions into RPR lib

yozhijk committed:
Merge pull request #166 from Kvalme/gltf_fixes

More GLTF related fixes

AlexanderVeselov committed:
Add materials.xml converter

AlexanderVeselov committed:
Replace std::map to std::unordered_map

AlexanderVeselov committed:
Apply gamma to textures

AlexanderVeselov committed:
Move gamma correction to different place

AlexanderVeselov committed:
Remove obj loading and unused code

AlexanderVeselov committed:
Remove comments from CMakeLists

yozhijk committed:
Merge pull request #165 from AlexanderVeselov/fix_embed_kernels

Fix embed kernels

Kvalme commented on Tools/MaterialConverter/CMakeLists.txt:
```diff
@@ -0,0 +1,33 @@
+    material_converter.cpp
+    material_converter.h
+    main.cpp
+    ${Baikal_SOURCE_DIR}/Baikal/Utils/half.cpp
```
Why you need this? You link with Baikal so you can use library version.

Kvalme commented on Tools/MaterialConverter/CMakeLists.txt:
```diff
@@ -0,0 +1,33 @@
+
+add_executable(MaterialConverter ${SOURCES} ${BAIKAL_OLD_SOURCES} ${BAIKAL_OLD_IO_SOURCES})
+target_compile_features(MaterialConverter PRIVATE cxx_std_14)
+target_include_directories(MaterialConverter PRIVATE 
```
I believe you don't need this since you link with Baikal

Kvalme commented on Tools/MaterialConverter/CMakeLists.txt:
```diff
@@ -0,0 +1,33 @@
+target_link_libraries(MaterialConverter PRIVATE Baikal BaikalIO)
+
+set_target_properties(MaterialConverter
+    PROPERTIES VS_DEBUGGER_WORKING_DIRECTORY ${Baikal_SOURCE_DIR})
```
Should it be launched from root or from /Tools/MaterialConverter ?
Does it really matter from where it will start?

Kvalme commented on Tools/MaterialConverter/material_converter.cpp:
```diff
@@ -0,0 +1,498 @@
+        LOG("BxdfType: Reflection");
+        uber_mtl->SetLayers(Baikal::UberV2Material::kReflectionLayer);
+        uber_mtl->SetInputValue("uberv2.reflection.color", TranslateInput(old_mtl, "albedo"));
+        uber_mtl->SetInputValue("uberv2.reflection.roughness", TranslateInput(old_mtl, "roughness"));
```
For ideal reflect you should set roughness to 0

Kvalme commented on Tools/MaterialConverter/material_converter.cpp:
```diff
@@ -0,0 +1,498 @@
+        break;
+    case BaikalOld::SingleBxdf::BxdfType::kPassthrough:
+        LOG("BxdfType: Passthrough");
+        // Nothing?
```
100% transparency

Kvalme commented on Tools/MaterialConverter/material_converter.cpp:
```diff
@@ -0,0 +1,498 @@
+        // Nothing?
+        break;
+    case BaikalOld::SingleBxdf::BxdfType::kTranslucent:
+        LOG("BxdfType: Translucensy");
```
Add assertion please. UberV2 don't support translucency. We got task to implement it.

Kvalme commented on Tools/MaterialConverter/material_converter.cpp:
```diff
@@ -0,0 +1,498 @@
+        LOG("BxdfType: Refraction");
+        uber_mtl->SetLayers(Baikal::UberV2Material::kRefractionLayer);
+        uber_mtl->SetInputValue("uberv2.refraction.color", TranslateInput(old_mtl, "albedo"));
+        uber_mtl->SetInputValue("uberv2.refraction.roughness", TranslateInput(old_mtl, "roughness"));
```
0.0f for Ideal

Kvalme commented on Tools/MaterialConverter/material_converter.cpp:
```diff
@@ -0,0 +1,498 @@
+    // Coating
+    {
+        bool base_has_coating = HasLayer(base_layers, Baikal::UberV2Material::kCoatingLayer);
+        bool top_has_coating = HasLayer(top_layers, Baikal::UberV2Material::kCoatingLayer);
```
Looks like it's always false.

Kvalme commented on Tools/MaterialConverter/material_converter.cpp:
```diff
@@ -0,0 +1,498 @@
+
+Baikal::UberV2Material::Ptr MaterialConverter::TranslateMix(BaikalOld::MultiBxdf::Ptr mtl)
+{
+    // CHECK: Might it be different from fresnel blend?
```
yes, it's different

AlexanderVeselov committed:
Fix issues from review

AlexanderVeselov committed:
Fix Linux errors

AlexanderVeselov committed:
Account for ior in TranslateSingleBxdfMaterial

Kvalme commented on Tools/MaterialConverter/material_converter.cpp:
```diff
@@ -0,0 +1,607 @@
+//#define USE_LOGFILE
+#ifdef USE_LOGFILE
+std::ofstream logfile("log.txt");
+#define LOG(x) logfile << x << std::endl
```
Replace with Baikal logging

AlexanderVeselov committed:
Use Baikal logging, fix translating passthrough bxdfs

AlexanderVeselov committed:
Remove half.cpp from MaterialConverter build files

AlexanderVeselov committed:
Minor fixes
- Skip SSS layer
+ Add common input maps for layers
+ Add check of existing materials.xml and mapping.xml
+ Fix help message
o Replace std::map kLayersInputs to std::vector

Kvalme commented on BaikalIO/material_io.cpp:
```diff
@@ -468,14 +486,14 @@ namespace Baikal
-                    std::ostringstream oss;
-                    oss << (std::uint64_t)texture.get() << ".jpg";
+                    //std::ostringstream oss;
+                    //oss << (std::uint64_t)texture.get() << ".jpg";
```
1) Remove dead code
2) It's ok for me to switch to names instead of adresses, but please make all names lower-case.

Kvalme commented on Tools/MaterialConverter/material_converter.cpp:
```diff
@@ -0,0 +1,592 @@
+    Baikal::InputMap::Ptr input_map = nullptr;
+    auto old_input_value = old_mtl->GetInputValue(input_name);
+    Baikal::LogInfo("Translating input \"", input_name.c_str(), "\" of material \"", old_mtl->GetName(), "\"\n");
+    switch (old_input_value.type)
```
please, add empty lines after each logical block in this file.

Kvalme commented on Tools/MaterialConverter/material_converter.cpp:
```diff
@@ -0,0 +1,592 @@
+    case BaikalOld::Material::InputType::kFloat4:
+        Baikal::LogInfo("Float4 value: (", old_input_value.float_value.x, ", ", old_input_value.float_value.y, ", ", old_input_value.float_value.z, ")\n");
+        input_map = Baikal::InputMap_ConstantFloat3::Create(old_input_value.float_value);
+        break;
```
new line after break

Kvalme commented on Tools/MaterialConverter/material_converter.cpp:
```diff
@@ -0,0 +1,592 @@
+    {
+        Baikal::LogInfo("Texture value: \"", old_input_value.tex_value->GetName(), "\"\n");
+        RadeonRays::int2 old_size = old_input_value.tex_value->GetSize();
+        char* data = new char[old_size.x * old_size.y];
```
new line before this line

Kvalme commented on Tools/MaterialConverter/material_converter.cpp:
```diff
@@ -0,0 +1,592 @@
+        RadeonRays::int2 old_size = old_input_value.tex_value->GetSize();
+        char* data = new char[old_size.x * old_size.y];
+        memcpy(data, old_input_value.tex_value->GetData(), old_size.x * old_size.y);
+        auto new_texture = Baikal::Texture::Create(data,
```
new line before this line

Kvalme commented on Tools/MaterialConverter/material_converter.cpp:
```diff
@@ -0,0 +1,592 @@
+            RadeonRays::int3(old_size.x, old_size.y, 1),
+            TranslateFormat(old_input_value.tex_value->GetFormat()));
+        new_texture->SetName(old_input_value.tex_value->GetName());
+        input_map = Baikal::InputMap_Sampler::Create(new_texture);
```
new line before this line

Kvalme replied:
And so on in entire file

Kvalme reviewed (changes requested):
Please, make material_converter.cpp don't look like wall of text

AlexanderVeselov committed:
Fix issues from review №2
+ Add formatting to the sources of MaterialConverter
+ Fix size of allocated array when converting textures from old version to new
+ Add saving of unnamed textures in MaterialIoXml

AlexanderVeselov committed:
Add some formalization
+ Add README
+ Add license to files

AlexanderVeselov committed:
Merge branch 'master' into material_converter
